src/global.interfaces.ts

```diff
@@ -0,0 +1,190 @@
+import { Document, Schema } from 'mongoose';
```

Comment:
Could we make a global/interface/ dir and have each one be its own file? This file is already getting unwieldy.